src/tasks.py

```diff
@@ -558,6 +562,47 @@ def get_metrics(self, reset=False):
         metrics['f1'] = f1
         return metrics
 
+# Entity type labeling on CoNLL 2003.
+@register_task('edges-ner-conll2003-sd', rel_path='edges/ner_conll2003',
+               label_file="labels.txt", files_by_split={
+                    'train': "CoNLL-2003_train.json" + _tokenizer_suffix,
+                    'val': "CoNLL-2003_dev.json" + _tokenizer_suffix,
+                    'test': "CoNLL-2003_test.json" + _tokenizer_suffix,
+               }, single_sided=True)
+# Entity type labeling on OntoNotes.
+@register_task('edges-ner-ontonotes-sd',
+               rel_path='edges/ontonotes-ner',
+               label_file="labels.txt", files_by_split={
+                    'train': "ner_ontonotes_en_train.json" + _tokenizer_suffix,
+                    'val': "ner_ontonotes_en_dev.json" + _tokenizer_suffix,
+                    'test': "ner_ontonotes_en_test.json" + _tokenizer_suffix,
+               }, single_sided=True)
+# SRL CoNLL 2012 (OntoNotes), formulated as an edge-labeling task.
+@register_task('edges-srl-conll2012-sd', rel_path='edges/srl_conll2012',
+               label_file="labels.txt", files_by_split={
+                    'train': "train.edges.json" + _tokenizer_suffix,
+                    'val': "dev.edges.json" + _tokenizer_suffix,
+                    'test': "test.edges.json" + _tokenizer_suffix,
+               }, is_symmetric=False)
+class SpanDetectionTask(EdgeProbingTask):
+    ''' Task class for edge probing
+
+    Same as EdgeProbingTasks but with span detection
+    '''
+    def __init__(self, path: str, max_seq_len: int,
+                 name: str,
+                 label_file: str=None,
+                 files_by_split: Dict[str,str]=None,
+                 is_symmetric: bool=False,
+                 single_sided: bool=False,
+    ):
+        max_seq_len = 200
```

Comment:
Just make `max_seq_len` just an argument in to the `@register_task` macro, instead of adding boilerplate here?

For efficiency, it might be worth running the [edge_data_stats.py](https://github.com/jsalt18-sentence-repl/jiant/blob/master/probing/edge_data_stats.py) script to get the max length for each task, then setting this explicitly.